src/utils/sidebar.js

```diff
@@ -45,36 +58,16 @@ function findItem(data, targetPath) {
   }
 }
 
-function findChildWithSource(item) {
-  return item.source ? item : findChildWithSource(item.children[0])
-}
-
-function findPrevItemWithSource(data, item) {
-  if (item.source) {
-    return item
-  } else if (item.prev) {
-    const prevItem = findItem(data, item.prev)
+function findPrevItemWithSource(data, ref) {
```

Comment:
why this rename - item to ref? is it consistent with other code around?

Reply:
The name `item` is used in other parts of this module for a different data structure. It was confusing when debugging, as these other areas of code call this function. When the argument is used, it's called `prevRef` in the caller, but the name of the function already has `Prev`, so I thought just `ref` would be appropriate here.

Reply:
could you give a bit more details? what is the difference among structures?

Reply:
Sure. basically `item` from the previous scope is a chunk of the sidebar.json struct, while `item` (now `ref` in this fn is a fully build "target structure" as spec'd on the top of the file.

<details>

`normalizeItem` calls this function sending `prevRef` to the argument in question, which is a previously "normalized" sidebar item:

https://github.com/iterative/dvc.org/blob/0b607e942be52160bb3429316663f360acec1be7/src/utils/sidebar.js#L78-L85

`normalizeItem` itself receives an `item` argument as well as `prevRef`. This scope's `item` comes from looping `data` in `normalizeSidebar`:

https://github.com/iterative/dvc.org/blob/0b607e942be52160bb3429316663f360acec1be7/src/utils/sidebar.js#L112-L120

which in turn is the original "raw" sidebar.json struct loaded when the server starts:

https://github.com/iterative/dvc.org/blob/0b607e942be52160bb3429316663f360acec1be7/src/utils/sidebar.js#L145-L148

</details>

Maybe `ref` isn't a great name either. In fact maybe this is the true `item` and some other `items` should be called `rawItem` instead. I can make that change, or I can just revert this.

Reply:
right. thanks! I think it's valuable indeed to distinct item and rawItem by name. It should be consistent around the file, variable names and function names. (like `validateRawItem` vs `findItem` for example)

Reply:
Done.

Let me clean up all the comments and work on the sidebar test. Then I'll make this a normal PR and ping you. Thanks!